python_runner/runner.py

```diff
@@ -35,6 +35,22 @@ def __init__(
     def set_callback(self, callback):
         self._callback = callback
 
+    def set_source_code(self, source_code, filename):
+        self.filename = os.path.normcase(os.path.abspath(filename))
+        # Write to file if permitted by system
+        try:
+            with open(self.filename, "w") as f:
+                f.write(source_code)
+        except: 
```

Comment:
```suggestion
        except:  # pragma: no cover
```

I'm assuming these lines aren't covered?

Reply:
Indeed. In a comment, you mentioned that writing a file should be unless used in a system where this is not possible, so I added a catch-all for that case. I don't know when this would occurr, so indeed these lines aren't covered.